### PR TITLE
Remove v2 search

### LINKED ISSUE #1899: Remove support for v2 search
- Cleanup and remove all flatbuffer code that is currently enabling v2 search. 
- Cleanup and remove all backend v2 search paths returning "not supported"

After 2.1 v2 will support trace by id lookup only. This feature was deprecated in 2.0.

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
As announced with the release of 2.0 v2 search will be removed for 2.1. This code got quite complicated and removing it will allow us to more confidently move forward with building TraceQL and additional features.

**Which issue(s) this PR fixes**:
Fixes #1899 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by joe-elliott** [comment]
@knylander-grafana pushed docs changes

**Review by zalegrala — APPROVED** [review]
Nice work, I love to see the quick turn around here.

File: docs/sources/tempo/operations/backend_search.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ some suggestions for tuning your backend to improve performance.
 General advice is to scale your compactors and queriers. Additional queriers can more effectively run jobs in parallel
 while additional compactors will more aggressively reduce the length of your blocklist and copies of data (if using RF=3).
 
+**Note:** All forms of search (TraceQL and tags based) are only supported on the `vParquet` and forward blocks. [v2 blocks]({{< relref "../configuration/parquet#disable-parquet" >}})
```
Comment (knylander-grafana) [inline docs/sources/tempo/operations/backend_search.md:15]: I've added the > to make this section formatted as an admonition block. Is there additional text missing around v2 blocks? 
```suggestion
>**Note:** All forms of search (TraceQL and tags based) are only supported on the `vParquet` and forward blocks. [v2 blocks]({{< relref "../configuration/parquet#disable-parquet" >}})
```

**Review by knylander-grafana — COMMENTED** [review]
Thank you for updating the docs! I've made a few suggestions.

**Pull request merged**